test/integration/test.js

```diff
@@ -427,6 +427,31 @@ describe('integration', function() {
       );
     });
 
+    it('should NOT catch an exception already caught [but rethrow because of transport error]', function(done) {
```

Comment:
i could imagine more test cases with 2 errors being sent through im quick succession, or transports that call opts.onError